src/main/groovy/edu/oregonstate/mist/catalogapitest/resources/CourseResource.groovy

```diff
@@ -1,42 +1,166 @@
 package edu.oregonstate.mist.catalogapitest.resources
 
+import edu.oregonstate.mist.catalogapitest.core.ErrorPOJO
 import edu.oregonstate.mist.catalogapitest.core.Course
 import edu.oregonstate.mist.catalogapitest.db.CourseDAO
 import io.dropwizard.jersey.params.IntParam
+import com.google.common.base.Optional
+import javassist.NotFoundException
+import org.eclipse.jetty.server.Response
+import javax.validation.Valid
 import javax.validation.constraints.NotNull
 import javax.ws.rs.Consumes
+import javax.ws.rs.DELETE
 import javax.ws.rs.GET
 import javax.ws.rs.POST
+import javax.ws.rs.PUT
 import javax.ws.rs.Path
 import javax.ws.rs.PathParam
 import javax.ws.rs.Produces
 import javax.ws.rs.QueryParam
 import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.Context
 import javax.ws.rs.core.MediaType
-import com.google.common.base.Optional
-import javassist.NotFoundException
-import org.eclipse.jetty.server.Response
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriInfo
 
 @Path("/course")
 @Produces(MediaType.APPLICATION_JSON)
 class CourseResource {
 
+    // Instantiate DAO -------------------------------------------------------------------------------------------------
     private final CourseDAO courseDAO
 
+    @Context
+    UriInfo uriInfo
+
     public CourseResource(CourseDAO courseDAO) {
         this.courseDAO = courseDAO
     }
 
+    // POST to /course ------------------------------------------------------------------------------------------------
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postUser(@Valid Course newCourse) {
+
+        Response returnResponse
+        def createdURI
```

Comment:
what is the type of `createdURI`? prefer explicitly typed variables whenever possible.